### DISCUSSION TITLE
Scoop - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#327] [Announcements]
# Scoop - CS自学指南

CS自学指南

https://csdiy.wiki/%E5%BF%85%E5%AD%A6%E5%B7%A5%E5%85%B7/Scoop/

**Comment by chen261014** [comment]
好东西。可惜我当年没碰到这样的东西，已经毕业快十年了。

**Reply by davidoffxw** [reply]
毕业已经超过十五年的我帮顶一下！

**Reply by white-keeper** [reply]
毕业已经超过六个月的我帮顶一下

**Reply by dapangtou** [reply]
今年第八年

**Comment by aixi** [comment]
这个域名好像被墙了，不用魔法是打不开的。

**Reply by 1202www** [reply]
我没有魔法能打开

**Reply by bomb076** [reply]
各家运营商的问题

**Reply by aixi** [reply]
应该是运营商的问题，有些运营商能打开

**Comment by jundecno** [comment]
大四才发现的宝藏

**Reply by decetion** [reply]
我大三才发现...还好现在其实也不晚，希望能坚持下去

**Comment by wwwkk** [comment]
非常支持，可惜我上学那会互联网信息还没现在这么发达，当时的我啥也不懂，也没能遇到前辈指点现在奔四的人了，来学习学习，非常感谢！

**Comment by VA020191001** [comment]
点赞

**Comment by Xbai-hang** [comment]
大四未来可期！梦开始的地方！

**Reply by suhu93** [reply]
大四加一

**Comment by fcjz** [comment]
想想还是在半年前使用Hugo搭建博客接触到了scoop，确实是好东西，方便好用，官方文档写的也很亲民了。

自己还是挺幸运的……离大学还有很长一段时间就接触到了这些，希望自己能够越来越好。

支持。

**Comment by asdukw** [comment]
庆幸自己大一发现了

**Reply by weqeweqewe** [reply]
大二才看到

**Reply by eve8776** [reply]
我也😭

**Reply by JzzzX** [reply]
大一下

**Comment by asdukw** [comment]
寒假刚看到，哈哈为什么这是用qq邮箱回复的。要加个微信吗（在github上交朋友或许很酷







------------------&nbsp;原始邮件&nbsp;------------------
发件人: "观 ***@***.***&gt;; 
发送时间: 2022年12月12日(星期一) 中午11:42
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [PKUFlyingPig/cs-self-learning] Scoop - CS自学指南 (Discussion #327)





 
大二才看到

**Reply by fcjz** [reply]
觉得交流的话其实邮箱之类更合适？

**Reply by Sunzijie** [reply]
邮箱交流有点高大上哈😂

**Comment by asdukw** [comment]
哈哈，可能是我们github上绑定的qq邮箱






------------------&nbsp;原始邮件&nbsp;------------------
发件人: "观 ***@***.***&gt;; 
发送时间: 2022年12月12日(星期一) 中午11:46
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [PKUFlyingPig/cs-self-learning] Scoop - CS自学指南 (Discussion #327)





 
我也不知道，就是邮箱一直消息哈哈哈哈，我是平时没事儿准备学cs，我本科是金融，微信18090175016

**Comment by 844843354** [comment]
针不戳，向你致敬，可惜我毕业了

**Comment by zhanyouzhou** [comment]
感谢捏

**Comment by myYouthlic** [comment]
- 问题
最后提出的下载 python27 的时候加入`bucket` 的命令错误， 应该改为
`scoop bucket add versions`

**Reply by suiyun39** [reply]
感谢反馈, 确实少了个 s, 今晚修复.

**Comment by Hurry1016** [comment]
unsubscribe

**Comment by zhengyaolaw** [comment]
谢谢您

**Comment by wing-py** [comment]
所以为啥不直接用linux呢？我大学四年用了四年ubuntu，从一开始ubuntu18.04LTS到ubuntu22.04LTS，过得很好，配置各种开发环境也没问题

**Reply by Sunzijie** [reply]
实用吗？曾经用过一星期，纯计算机专业还可以，微信QQ的Linux版也还能用，office功能不全，想折腾ps pr
就没辙了，用插件也可以安装使用windows软件，终归水土不服。备用机的话装个Linux还不错 o((⊙﹏⊙))o.

**Reply by ulic-youthlic** [reply]
如果是没有特殊需求的话，搞个双系统也挺好。

**Reply by ulic-youthlic** [reply]
实际上 arch 的体验还是可以的，UI 和下拉终端都可以的，office 也有平替，只是 ps 这类专业软件没辙

**Reply by ScSofts** [reply]
> 实际上 arch 的体验还是可以的，UI 和下拉终端都可以的，office 也有平替，只是 ps 这类专业软件没辙

office只有底替没有平替，wps转换个pdf都要开会员，逆向过发现要走服务器没法破解。。。

你用word打开一份pdf文档试试看，就再也离不开了。更别说还有“听写”、“贴纸”这类高级功能。至于自带的那个什么开源office，ppt做出来在Windows上放会走样，而且界面还是07年的风格。

反而是ps，倒是有gimp这种东西。

Linux最烦的就是什么事都得敲命令，新建个文件还要打开终端。

**Reply by ScSofts** [reply]
开发环境，为什么不用docker。跨平台支持，随便搞不可能水土不服

**Comment by star-Triangel** [comment]
大三发现的宝藏。谢谢学长。

**Comment by Sunzijie** [comment]
求助：想知道图标的含义 和底下一串英文

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by meng-2003** [comment]
爱了爱了

**Comment by FJianC** [comment]
切换国内镜像:
1. 安装脚本（已安装的跳过）：`iwr -useb https://gitee.com/glsnames/scoop-installer/raw/master/bin/install.ps1 | iex`
2. 修改镜像：`scoop config SCOOP_REPO https://gitee.com/glsnames/scoop-installer`
3. bucket镜像：
   - 新加bucket：`scoop bucket add main https://gitee.com/scoop-bucket/main.git`
   - 修改已加的bucket：
   - `cd $env:SCOOP\buckets\main`
   - `git remote set-url origin https://gitee.com/scoop-bucket/main.git`

切换回官方镜像:
1. `scoop config SCOOP_REPO https://github.com/ScoopInstaller/Scoop`
2. 重新添加已有的bucket：
   - `scoop bucket rm main`
   - `scoop bucket add main`

**Reply by liuyizhe** [reply]
这东西是不是和mac上的brew对等的

**Comment by Murlors** [comment]
`winget`现在的体验也挺不错了，`Windows 10 1709`以上的系统都可以使用。\
<https://learn.microsoft.com/zh-cn/windows/package-manager/winget/>

**Reply by ulic-youthlic** [reply]
`winget` 最重要的是不能自定义默认安装位置，如果每次都要手动 `-l` 还是很麻烦的

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by ScSofts** [comment]
大家知道不，其实Win11有winget

**Reply by ulic-youthlic** [reply]
无所谓，我用scoop

**Reply by melodylx666** [reply]
thanks

**Reply by ghost** [reply]
难用

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Reply by Wh1tew0lf-sea** [reply]
好！休假！好！

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by JichiGe** [comment]
如果想要用scoop 安装numbpy matplotlib咋操作啊

**Reply by ulic-youthlic** [reply]
如果你说的是 `numpy matplotlib` 的话，似乎没有什么办法（？
-------
以下是支持该观点的一些理由：
### 在 `scoop` 的包 [搜索网站](https://scoop.sh) 搜索的结果
  - 搜索 `numpy matplotlib`
------   
![image](https://user-images.githubusercontent.com/121918198/235373297-bcf302b0-ae30-48ef-aabf-896a91c9e446.png)
------
  - 搜索 `numpy`
------
![image](https://user-images.githubusercontent.com/121918198/235373308-ebce6060-9971-4a7c-836e-0165418479e7.png)
------
  - 搜索 `matplotlib`
------
![image](https://user-images.githubusercontent.com/121918198/235373314-37ba193e-d032-4845-b280-be0e65025ab2.png)

**Reply by cscnk52** [reply]
numpy matplotlib 是 python 中的包，需要通过 python 的包管理器进行下载安装， 不属于 scoop 的范围
```bash
scoop install python
pip install numpy matplotlib
```
可通过上面两条命令安装

**Reply by JichiGe** [reply]
谢了啊， 但是 scoop 安装的地址和pip的默认地址很可能不一致吧？所以需要手动把pip的安装地址改成scoop的然后安装的，对吗

**Reply by JichiGe** [reply]
你是怎么操作的啊

**Reply by cscnk52** [reply]
你去网上搜关于 python 环境的教程，都是下安装包、安装、加环境变量，而 scoop 就是自动帮你把这些任务给完成了，之后的下 numpy 包就和 scoop 没有关系了！！！如果还是不清楚，建议搜索 Ubuntu 的 apt 、 MAC OS 的 Homebrew 或 Arch Linux 的 pacman 了解，或者理解为 命令行里的应用商店。

至于你说的两个包都是 python 中的，需要通过 pip 进行安装，不清楚看这个链接 https://www.runoob.com/w3cnote/python-pip-install-usage.html

再附上两个包的链接
- https://pypi.org/project/matplotlib/
- https://pypi.org/project/numpy/

再回答你提出的问题，两个地址肯定是不一样的，一个是管理 python 中的包，一个是管理 Windows 中的程序，两者管理的范围是不一样的，也无从谈起改地址。

但改地址的情况可以发生在切换为镜像源，具体方法查询开源软件镜像站的相关帮助页面。

以上问题回答看不懂的请先搜索，关于如何提问请查看这个链接 https://github.com/ryanhanwu/How-To-Ask-Questions-The-Smart-Way/blob/main/README-zh_CN.md

**Comment by blameitonme1** [comment]
真的庆幸我知道了这个工具，简直是太好用了，环境变量之类的也直接配好，以前配环境花很多时间，现在只需要命令行输入几行命令就行，非常实用的工具！感谢分享

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by meowrain** [comment]
```
scoop config proxy 127.0.0.1:2080
```

scoop安装软件的话，直接配置一下它的代理就好了，端口是本地的代理软件端口

**Reply by meowrain** [reply]
# 安装scoop
```powershell
> Set-ExecutionPolicy RemoteSigned -Scope CurrentUser # Optional: Needed to run a remote script the first time
> irm get.scoop.sh | iex
```

# 更新软件库
```powershell
scoop bucket add extras
scoop bucket add versions

scoop update
```

# 配置scoop代理

> 国内网络环境原因，这样是最佳解决办法
```powershell
scoop config proxy 127.0.0.1:2080
```


# 安装常用环境/工具

java
`https://scoop.sh/#/apps?q=openjdk`

mingw-64(c/c++环境)
```powershell
scoop install main/mingw
```

go
```powershell
scoop install main/go
```

rust
```powershell
scoop install main/rustup-gnu
```

python
`https://scoop.sh/#/apps?q=python`
```powershell
scoop install main/python
```

git
```powershell
scoop install main/git
```

adb-tools(安卓开发必备)
```powershell
scoop install main/adb
```

nvm（推荐)
```powershell
scoop install main/nvm
```

nodejs
```powershell
scoop install main/nodejs-lts
```

maven
```powershell
scoop install main/maven
```


mysql
`https://scoop.sh/#/apps?q=mysql`

```powershell
scoop install extras/heidisql
scoop install main/mysql-lts
```

vscode
```powershell
scoop install extras/vscode
```

sublime
```powershell
scoop install extras/sublime-text
```

idea
```powershell
scoop install extras/idea-ultimate
```

pycharm
```powershell
scoop install extras/pycharm-professional
```

webstorm
```powershell
scoop install extras/webstorm
```


steam
```powershell
scoop install games/steam
```


chrome
```powershell
scoop install extras/googlechrome
```

firefox
```powershell
scoop install extras/firefox
```

**Comment by AgentOfLyra** [comment]
现在有些东西可以直接在开发人员主页里面下载了

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by stepbystepcode** [comment]
感觉winget和choco也不错

**Reply by Qxy0happy** [reply]
choco权限请求不太合理，winget有的开发库还是不太齐全

**Comment by RatRat0** [comment]
Scoop的安装可能会遇到终端无法翻墙的问题，照着这个文章来装亲测效果很好，https://zhuanlan.zhihu.com/p/678607608，顺便也把环境也给配置了😊

**Reply by RatRat0** [reply]
终端能翻墙了以后一些安装源的问题也不用考虑了🤭

**Reply by yin127ing** [reply]
有计算机组成原理的吗

**Reply by pingtougea** [reply]
大佬，配终端需要代理，纯小白不知道咋搞

**Comment by yin127ing** [comment]
有计算机组成原理的课程吗

**Reply by pigziv** [reply]
你往下翻应该会有的

**Comment by kaerljnd** [comment]
CS61A需要登录伯克利怎么做？

**Reply by ulic-youthlic** [reply]
学习往期课程，而不是最新的版本

**Comment by pigziv** [comment]
想问下如何翻墙啊啊啊啊，特别想自学计算机，但是最开始入门就被卡住了qaq，找了很多免费节点都特别的卡，有没有大佬能够分享一下好一点的给我啊啊啊啊啊。

**Reply by RatRat0** [reply]
链接：https://pan.baidu.com/s/1u7uL5MghDGSuulZPX98xCg?pwd=vzgz 
提取码：vzgz
因为外网的东西在内网没法查，所以放网盘里了

**Reply by RatRat0** [reply]
好吧，不知道怎么的，发不了

**Reply by pigziv** [reply]
ooOOO那怎么发，感谢大佬

**Reply by pigziv** [reply]
他显示链接不存在

**Reply by ulic-youthlic** [reply]
用onedrive试试？百度会扫盘

**Reply by RatRat0** [reply]
2930204992加我QQ吧，我直接发文件给你

**Reply by Unrequired** [reply]
可以試試看使用warp(1.1.1.1)，在B站上有教學如何得到warp+

**Reply by teacup418** [reply]
[魔法上网](https://blog.friesport.ac.cn/articles/25/)，密码proxy，请你自己偷偷看，感谢。

**Comment by pangxuancong** [comment]
大家是怎么学习的，是又看slide又看教材然后看回放还是只看slide和回放？

**Comment by caohuisheng** [comment]
非常好的资源，要是我早点遇到就好了！

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by jinnyi-l** [comment]
硕士才转码，浏览网页发现宝藏wiki，对我帮助太大了，后来才发现居然是国人写的，难怪我水土这么服，万分感激，祝作者生活一切顺利！

**Comment by Sqyl** [comment]
Love it, 今年研一，码住准备从头开始

**Comment by previet** [comment]
大一非计院专业正在自学，想找个学习搭子互相监督

**Comment by Magicdover** [comment]
那么，开始吧

**Comment by zhangbird050801** [comment]
英文课程可以搭配 Chrome 插件 Trancy，还有 沉浸式翻译插件。

**Comment by HITLittleZheng** [comment]
好东西啊！可惜没及时发现，大三前几年光为了保研卷学分绩了，现在都已经大三下了，发现自己跟别人差太远了

**Comment by unibeam98** [comment]
有winget可以用来参考

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by KENNY-hub-vue** [comment]
请问有人碰到安装 versions 这个 bucket 报错的吗？有的话有无解决的帖子借鉴一下，在搜索之后暂未找到合适的解决办法，谢谢。

**Reply by teacup418** [reply]
你好，请问你开魔法上网了吗？可以提供更详细的报错吗？理论上versions都是过时的版本，应该是不需要添加的。

**Reply by KENNY-hub-vue** [reply]
您好，就是我想着要装 python2.x ,具体的报错等我找一下哈，我正在重新下载 versions 这个 bucket。
运行的结果整个如下：
❯ scoop bucket add versions
Checking repo... error: RPC failed; curl 56 Failure when receiving data from the peer
error: 873 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
OK
The versions bucket was added successfully.

**Reply by KENNY-hub-vue** [reply]
哦，对了，我记得我应该是开了代理。

**Reply by teacup418** [reply]
看起来你的代理的网络状况不太好，ping一下看看节点通不通，不通的话换个节点，或者等待一个网络状况良好的时候再尝试。

**Reply by KENNY-hub-vue** [reply]
感谢，我重新断开了代理之后可以了，可能一开始想着要尝试断代理，没做就重启终端了，感谢。

**Comment by KENNY-hub-vue** [comment]
好的，昨天太累了就先睡了，非常感谢您的帮助。我今天去学车回来之后再尝试一下，🌹🌹



---Original---
From: "Vincent ***@***.***&gt;
Sent at: 2024年6月22日(Sat) Early Morning0:52
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [PKUFlyingPig/cs-self-learning] Scoop - CS自学指南 (Discussion #327)




 
看起来你的代理的网络状况不太好，ping一下看看节点通不通，不通的话换个节点，或者等待一个网络状况良好的时候再尝试。

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by Qxy0happy** [comment]
scoop有国内源可以使用：scoop bucket add scoop-cn https://ghp.ci/https://github.com/duzyn/scoop-cn

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by KENNY-hub-vue** [comment]
邮件已收到，谢谢你🌹



---Original---
From: ***@***.***&gt;
Sent at: 2024年12月15日(Sun) Noon12:11
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [PKUFlyingPig/cs-self-learning] Scoop - CS自学指南 (Discussion #327)




 
scoop有国内源可以使用：scoop bucket add scoop-cn https://ghp.ci/https://github.com/duzyn/scoop-cn

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by bbbkawaii** [comment]
机械类大一,准备开始学习

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by ChengBoXu666** [comment]
直接用wsl会不会更好呢

**Reply by Qxy0happy** [reply]
也可以，我就是这么做的，配置环境比折腾Windows要方便不少

**Comment by fall-zhang** [comment]
有类似的应用，Chocolatey，我一直是在用这个，也写了一篇笔记关于[前端环境搭建](https://fallzhang.top/blog/2%E5%88%86%E9%92%9F%E7%8E%AF%E5%A2%83%E6%90%AD%E5%BB%BA)。希望对大家有帮助

**Comment by qingzhixing** [comment]
windows chocolatey也好用，PowerShell 下载: Set-ExecutionPolicy Bypass -Scope Process -Force; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))

**Reply by Qxy0happy** [reply]
chocolatey的权限模型问题太大了，随便做点什么都要管理员权限。

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by lil-Journey** [comment]
自建日本节点服务器 月租5元不限流量 本人也在用 失效时可联系人工修复 qq:1165116287

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by Sculpture112** [comment]
非常感谢您

**Reply by ywh555hhh** [reply]
真的是很好的学习资源！

如果有人一起来学习讨论就好了，我是这么想的。自学的过程中如果有伙伴，或者有人交流一定更棒。我想一定有人也有类似的需求，所以我和伙伴们创建了一个 [discord 服务器](https://discord.gg/Rux6DHRStP)，不知道大家有没有兴趣来看看！

如有不妥当请联系删除

**Comment by ywh555hhh** [comment]
真的是很好的学习资源！

如果有人一起来学习讨论就好了，我是这么想的。自学的过程中如果有伙伴，或者有人交流一定更棒。我想一定有人也有类似的需求，所以我和伙伴们创建了一个 [discord 服务器](https://discord.gg/Rux6DHRStP)，不知道大家有没有兴趣来看看！

如有不妥当请联系删除

**Comment by luc-fei** [comment]
好东西，金融跨考了大数据专硕，数据开发大厂实习中，感觉只会写SQL纯SQL-Boy，其他底层一点不懂，希望现在发现的不晚！！！

**Comment by L20wang** [comment]
还有winget和chocolately，现在有个图形化APP叫Uniget把常见的win包管理器整合了，很好用

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by bet178** [comment]
我小学生，高血压，甲亢，脑鸣，46岁有编程梦想。70年代人坚持的意义是什么？

**Comment by Harry-Yu-Shuhang** [comment]
现在是不是用winget好一些，原生的

**Comment by white-keeper** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by Qxy0happy** [comment]
winget只是代替你安装了，对GUI可能方便点，但是CLI还是scoop管理方便。
